data/scripts.js

```diff
@@ -378,8 +378,12 @@ let BattleScripts = {
 		const hitResults = this.runEvent('Invulnerability', targets, pokemon, move);
 		for (const [i, target] of targets.entries()) {
 			if (hitResults[i] === false) {
-				if (!move.spreadHit) this.attrLastMove('[miss]');
-				this.add('-miss', pokemon, target);
+				if (move.smartTarget) {
+					move.smartTarget = false;
+				} else {
+					if (!move.spreadHit) this.attrLastMove('[miss]');
```

Comment:
I actually like my implementation since it can also be used to `attrLastMove('[miss]')` if all hits of a spread hit miss.

Reply:
What's correct behaviour for a mixture of move failures e.g. a Protect, a Flying type immunity and an evasion miss while attempting an Earthquake?

Reply:
If it's not Dragon Darts, all the messages should appear, I think.